pkg/lifecycle/binary.go

```diff
@@ -132,32 +133,52 @@ func DownloadBinary(url, destDir, destFile string, logger *zap.Logger) (string,
 		}
 	}()
 
-	logger.Info("downloading binary", zap.String("url", url))
+	logger.Info("downloading binary", zap.String("url", sourceURL))
 
-	req, err := http.NewRequest("GET", url, nil)
+	u, err := url.Parse(sourceURL)
 	if err != nil {
-		return "", fmt.Errorf("could not create request (%w)", err)
+		return "", fmt.Errorf("could not parse URL %s (%w)", sourceURL, err)
 	}
-	client := &http.Client{}
-	resp, err := client.Do(req)
-	if err != nil {
-		return "", fmt.Errorf("HTTP GET %s failed (%w)", url, err)
-	}
-	defer resp.Body.Close()
 
-	if resp.StatusCode != http.StatusOK {
-		return "", fmt.Errorf("HTTP GET %s failed with error %d", url, resp.StatusCode)
-	}
+	switch u.Scheme {
+	case "http", "https":
```

Comment:
[ASK] Are we gonna support `http` too? or only `https`?

Reply:
No we don't, just place http there to make testing process simpler 🙏